### ISSUE TITLE
Docker:Vermittlungscode unvollständig eingetragen

### ISSUE THREAD

**Issue by lucaleicht** [TobseF/impf-bot#59]
Hallo zusammen,
nachdem ich bei der Desktop-Installation an einem unerklärlichen Java-Problem gescheitert bin, habe ich es jetzt mit Docker versucht.
Grundsätzlich bin ich hier schon deutlich weiter. Allerdings werden teilweise die Vermittlungscodes nicht vollständig eingetragen. Mal nur der erste Block, mal fehlen die letzten zwei Ziffern/Lettern. Manchmal fehlen in der Mitte Teile des Codes. Manchmal wird überhaupt kein Code eingetragen. Alles total willkürlich. Grundsätzlich arbeitet sich das System aber regulär durch.

Vielleicht ist dieses Problem ja lösbarer als das letzte ;)

**Comment by lucaleicht** [comment]
Vermutlich das 5 & 6-Problem mit einem alten Docker Image. Ich versuche mal ein Update.

Edit: Update erfolglos. Außerdem wird auch die "2" nicht übernommen. Auch ein "C" hat es einmal nicht geschafft, obwohl im es im selben Code einen Block vorher übernommen wurde.

**Comment by Pfuenzle** [comment]
Also das 5 und 6 Problem liegt am docker Image, ich bin noch nicht dazugekommen, das Dockerimage auf Dockerhub hochzuladen, da das bei meinem Internet ewig dauert, also müsstest du das Image kurz selber bauen oder bei heute Mittag warten. 
Die anderen Fehler kommen mir allerdings unbekannt vor, ich hab ansonsten eigentlich alle möglichen Tasten mir Firefox, Chrome und Edge ausprobiert

**Comment by Bergdoc** [comment]
Komischerweise existiert das 5 und 6 Problem immer noch @Pfuenzle 
Ich hatte das Image heute morgen kurz gebaut und getestet und es lässt sich jetzt auch die 5 und 6 über den VNC Client z.B. direkt händisch in die Vermittlungscode Zeile eintragen, aber das automatische ausfüllen scheitert immer noch.
Ich teste es aber heute Abend nochmal etwas mehr im Detail.

**Comment by lalelunet** [comment]
Ich habe das auch mit der aktuellen docker version. Version 1.2 habe ich getestet. Dort tritt es auch auf. Der Vermittlungscode ist aber nicht meiner? Er stimmt überhaupt nicht überein. Er bricht immer an der gleichen Stelle ab und ist immer identisch.

Was mir auch aufgefallen ist das der Server ingoriert wird. Der request wurde an 001 gesendet obwohl ich (002) angehängt habe.
```
[XXXX-XXXX-XXXX](002)
```
<img width="622" alt="Bildschirmfoto 2021-05-19 um 18 16 40" src="https://user-images.githubusercontent.com/4786404/118850616-42017700-b8d1-11eb-9085-55c5f9c3b616.png">
<img width="947" alt="Bildschirmfoto 2021-05-19 um 18 21 33" src="https://user-images.githubusercontent.com/4786404/118852203-da4c2b80-b8d2-11eb-9105-a5b710f1e151.png">

**Comment by mafeko** [comment]
@Pfuenzle hättest du Interesse das docker image jeweils mit dem aktuellen commit-hash als tag von github-actions bauen zu lassen? 
z.b.: pfuenzle/impf-bot:1b0995e
Sollte bei der anfallenden Nutzung kostenlos bleiben (siehe [docs](https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#:~:text=Workflow%20run%20time%20%2D%20Each%20workflow,might%20cause%20jobs%20to%20fail.))

**Comment by Pfuenzle** [comment]
Ja, werde ich ab jetzt machen. Hatte bis jetzt vor, einfach nur den :latest tag zu bauen, v1.2 etc waren nur noch verfügbar ,weil ich  die Dokumentation von Dockerhub missverstanden hatte und dachte, dass man Images nur mit Pro löschen kann. Hab die alten jetzt gelöscht und le die Images sollten ab jetzt automatisch gebuildet mit :latest und :SHAhash gebaut werden

**Comment by mafeko** [comment]
Okay cool, komme zwar nicht aus der Java Ecke aber könnte ggf. bei der Docker CICD pipeline unterstützen wenn du möchtest. ;)

**Comment by Sebastian-G** [comment]
> Okay cool, komme zwar nicht aus der Java Ecke aber könnte ggf. bei der Docker CICD pipeline unterstützen wenn du möchtest. ;)

Kannst ja einfach n MR erstellen 😉 dann muss @TobseF nur die credentials noch nachtragen.

**Comment by EQ5** [comment]
Kann mir kurz jemand erklären, wie ich das Problem behebe mit dem Unvollständige Vermittlungscode? Danke

**Comment by Sebastian-G** [comment]
Du kannst docker build starten um ein neues lokales image zu erzeugen.


1.  `docker build . -t custom-local-impf-bot` see https://github.com/TobseF/impf-bot/blob/master/doc/docker/DOCKER_SETUP.md

2.  change the used image on `docker-compose.yml` 
```
# OLD        image: pfuenzle/impf-bot
# NEW       image: custom-local-impf-bot
```

**Comment by pauli2406** [comment]
Hatte heute das selbe Problem und wollte es lösen, in dem ich das Image selber baue.
Allerdings tritt dann bei mir das folgende Problem auf:
![image](https://user-images.githubusercontent.com/24795790/120218351-e4a6e780-c239-11eb-9ad7-9ab2c0cd4378.png)

Er kommt gar nicht erst zu der Seite sondern gibt immer data; ein..
Jemand eine Idee?

**Comment by anp369** [comment]
Schau mal in die Konsole obs Exceptions oder so gibt. Hatte das Problem mal als mir vim Leerzeichen in mein config file gesetzt hat. Schau mal ob zb das Alter Leerzeichen am Ende hat. Das hatte das Problem zumindest bei mir immer behoben
